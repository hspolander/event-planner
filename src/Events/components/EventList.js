import React, { useEffect } from 'react';
import { List, ListSubheader } from '@material-ui/core';
import { connect } from 'react-redux';

import EventItem from './EventItem';
import Loading from './Loading';
import Error from './Error';
import { eventListType, errorType, loadingType, fetchEventsType } from '../../types';
import { fetchEvents } from '../actions';

const EventList = ({ events, loading, error, fetchEvents }) => {
  useEffect(() => {
    fetchEvents();
  }, [fetchEvents]);
  return (
    <List
      aria-labelledby="events-subheader"
      subheader={
        <ListSubheader component="div" id="events-subheader">
          Events
        </ListSubheader>
      }
    >
      {loading ? (
        <Loading loading={loading} />
      ) : error ? (
        <Error error={error} />
      ) : events ? (
        events.map((event, index) => <EventItem key={index} event={event} />)
      ) : null}
    </List>
  );
};

EventList.propTypes = {
  events: eventListType,
  loading: loadingType,
  error: errorType,
  fetchEvents: fetchEventsType,
};

const mapStateToProps = ({ eventReducer }) => ({
  events: eventReducer.events,
  loading: eventReducer.loading,
  error: eventReducer.error,
});
const mapDispatchToProps = { fetchEvents };

export default connect(mapStateToProps, mapDispatchToProps)(EventList);
