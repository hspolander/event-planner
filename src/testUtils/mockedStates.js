export const emptystate = {
  eventReducer: { events: null, fetching: false, error: false },
};

export const stateWithEvents = {
  eventReducer: {
    fetching: false,
    error: false,
    events: [
      {
        id: 3,
        startDate: '2019-02-26 10:00',
        endDate: '2019-02-26 12:00',
        location: 'Lagerhuset',
        activity: 'meeting',
        date: '2019-02-26 10:00:00',
      },
      {
        id: 1,
        startDate: '2019-02-26 12:00',
        endDate: '2019-02-26 13:00',
        location: 'Lagerhuset',
        activity: 'lunch',
        date: '2019-02-26 12:00:00',
      },
      {
        id: 2,
        startDate: '2019-02-26 22:00',
        endDate: '2019-02-28 07:00',
        location: 'online',
        activity: 'meeting',
        date: '2019-02-26 22:00:00',
      },
      {
        id: 2,
        startDate: '2019-02-26 22:00',
        endDate: '2019-02-28 07:00',
        location: 'online',
        activity: 'meeting',
        date: '2019-02-27 00:00:01',
      },
      {
        id: 2,
        startDate: '2019-02-26 22:00',
        endDate: '2019-02-28 07:00',
        location: 'online',
        activity: 'meeting',
        date: '2019-02-28 00:00:01',
      },
    ],
  },
};

export const stateWithError = {
  eventReducer: { events: null, fetching: false, error: true },
};

export const stateWhileLoading = {
  eventReducer: { events: null, fetching: true, error: false },
};
