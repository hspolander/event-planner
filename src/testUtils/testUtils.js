import React from 'react';
import { render } from '@testing-library/react';
import thunk from 'redux-thunk';
import { createStore, applyMiddleware } from 'redux';
import { Provider } from 'react-redux';
import rootReducer from '../rootReducer';

export default function renderWithRedux(
  ui,
  { initialState, store = createStore(rootReducer, initialState, applyMiddleware(thunk)) } = {}
) {
  return {
    ...render(<Provider store={store}>{ui}</Provider>),
    store,
  };
}

export * from '@testing-library/react';
