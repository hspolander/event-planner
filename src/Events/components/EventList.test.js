import React from 'react';
import renderWithRedux, { cleanup } from '../../testUtils/testUtils';
import { stateWithEvents } from '../../testUtils/mockedStates';

import EventList from './EventList';

describe('EventList', () => {
  afterEach(() => {
    cleanup();
  });

  it('should match snapshot', () => {
    const { container } = renderWithRedux(<EventList />, {
      stateWithEvents,
    });
    expect(container).toMatchSnapshot();
  });
});
