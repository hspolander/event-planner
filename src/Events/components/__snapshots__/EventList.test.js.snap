// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventList should match snapshot 1`] = `
<div>
  <ul
    aria-labelledby="events-subheader"
    class="MuiList-root MuiList-padding MuiList-subheader"
    role="list"
  >
    <div
      class="MuiListSubheader-root MuiListSubheader-sticky MuiListSubheader-gutters"
      id="events-subheader"
    >
      Events
    </div>
  </ul>
</div>
`;
